spynnaker/pyNN/models/neuron/population_machine_vertex.py

```diff
@@ -627,3 +628,35 @@ def gen_on_machine(self):
     def read_generated_connection_holders(self, transceiver, placement):
         self._app_vertex.synapse_manager.read_generated_connection_holders(
             transceiver, placement)
+
+    @overrides(AbstractReadParametersBeforeSet.read_parameters_from_machine)
+    def read_parameters_from_machine(
+            self, transceiver, placement, vertex_slice):
+
+        # locate SDRAM address to where the neuron parameters are stored
+        neuron_region_sdram_address = self.neuron_region_sdram_address(
+            placement, transceiver)
+
+        # shift past the extra stuff before neuron parameters that we don't
+        # need to read
+        neuron_parameters_sdram_address = (
+            neuron_region_sdram_address +
+            self._app_vertex.tdma_sdram_size_in_bytes +
+            self._app_vertex.BYTES_TILL_START_OF_GLOBAL_PARAMETERS)
+
+        # get size of neuron params
+        size_of_region = self._app_vertex.get_sdram_usage_for_neuron_params(
+            vertex_slice)
+        size_of_region -= (
+            self._app_vertex.BYTES_TILL_START_OF_GLOBAL_PARAMETERS +
+            self._app_vertex.tdma_sdram_size_in_bytes)
+
+        # get data from the machine
+        byte_array = transceiver.read_memory(
+            placement.x, placement.y, neuron_parameters_sdram_address,
+            size_of_region)
+
+        # update python neuron parameters with the data
+        self._app_vertex.neuron_impl.read_data(
+            byte_array, 0, vertex_slice, self._app_vertex.parameters,
+            self._app_vertex.state_variables)
```

Comment:
This is the ultimate call of all the app vertex stuff... any way this could come down here too?

Reply:
> I've had a look at this. My main issue is that all the machine vertex calls do is call the app vertex lots. Is this by choice, or is this by definition something that is happening as a midway point?

def a midway point. coz we dont know how olivers stuff is going to impact it. The current thinking is that the APp vertex is a DATA holder, and so neuron impl is safe staying up there. Given the different machine verts impl will still need the neuron params at some point. it seems sensible to keep that at the app level to avoid cloning code. Esp as the call now takes the dsg region as a input param, to allow the flex to the neuron core to put it whereever it wants to

Reply:
Yes, I think I was just hopeful.  I guess there isn't really an issue with the sharing, so long as the route to the data is consistent.